utils/retrieve-state/src/main.rs

```diff
@@ -217,14 +239,40 @@ async fn main() -> Result<(), anyhow::Error> {
             );
         }
         Action::DownloadTries => {
-            let mut client = ClientBuilder::new().gzip(opts.use_gzip).build().unwrap();
+            let client = ClientBuilder::new().gzip(opts.use_gzip).build().unwrap();
+
+            let mut peers_list = retrieve_state::get_peers_list(&client, &url)
+                .await?
+                .peers
+                .into_inner()
+                .into_iter()
+                .flat_map(|peer| {
+                    let address = peer.address.parse::<SocketAddrV4>().ok()?;
+                    // for nctl-networks we want our bound address
+                    let address = if port_derived_from_peers {
+                        SocketAddr::V4(SocketAddrV4::new(
+                            *address.ip(),
+                            address.port().saturating_sub(11000),
+                        ))
+                    } else {
+                        // the peer's list gives us the port for the node, and instead we want to
+                        // use the user-provided RPC port
+                        SocketAddr::V4(SocketAddrV4::new(*address.ip(), port))
+                    };
+                    Some(address)
+                })
+                .collect::<Vec<_>>();
+
+            // also use the server we provided inititially that peers were gathered from.
+            peers_list.push(initial_server_address);
+
             let highest_block: JsonBlock =
-                retrieve_state::get_block(&mut client, &url, maybe_download_block)
+                retrieve_state::get_block(&client, &url, maybe_download_block)
                     .await?
                     .block
-                    .unwrap();
-            let progress = ProgressBar::new(highest_block.header.height);
-            println!(
+                    .expect("unable to download highest block");
+
+            eprintln!(
```

Comment:
I suggest that this code avoid trusted RPC calls and instead require a trusted hash.  It'd then get blocks to current using finality signatures to verify them.

Reply:
You can do this now since the `highest_block` parameter is a `BlockIdentifier`. I'll add a note that asking for the highest block by height is considered insecure.

Reply:
I'd still prefer an `--unsafe` so they have to acknowledge they are doing something dangerous... like `kill -9` or `sudo`.

As discussed there is a security protocol for getting the highest block.  I'll see if I can quick, prototype something for you...